aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp

```diff
@@ -0,0 +1,210 @@
+#ifdef USE_CUDA
+#include <ATen/cuda/CUDAConfig.h>  // for the definition of AT_CUDNN_ENABLED
+
+#if AT_CUDNN_ENABLED()
+#include <ATen/native/cudnn/Macros.h>
+#if HAS_CUDNN_V8()
+
+#include <ATen/core/TensorBase.h>
+#include <ATen/core/TensorBody.h>
+#include <ATen/cuda/Exceptions.h>
+#include <ATen/cudnn/Handle.h>
+#include <ATen/native/quantized/cudnn/utils.h>
+#include <ATen/native/utils/ParamsHash.h>
+#include <ATen/TensorUtils.h>
+#include <c10/core/QScheme.h>
+#include <torch/library.h>
+
+#include <unordered_map>
+
+
+struct CacheKey {
+  uint8_t input_a_alignment;
+  uint8_t input_b_alignment;
+  uint8_t output_alignment;
+};
+
+// FIXME: make this thread-safe by reusing the benchmark cache in Conv_v7.cpp
+namespace {
+std::unordered_map<CacheKey, cudnn_frontend::ManagedOpaqueDescriptor, at::native::ParamsHash<CacheKey>, at::native::ParamsEqual<CacheKey>> execution_plan_cache;
+}
+
+namespace at {
+namespace native {
+namespace {
+// TODO: this is also in qadd.cpp and some other cpp files in quantized/cpu/. I think we should
+// move everything into a utilities file in quantized/ directory later.
+inline void check_inputs(const Tensor& qa, const Tensor& qb) {
+  TORCH_CHECK(
+      qa.qscheme() == kPerTensorAffine,
+      "Only per tensor quantization is suported in Add.");
+  TORCH_CHECK(
+      qa.qscheme() == qb.qscheme(),
+      "Both inputs to Add must have the same quantization shceme.");
+  TORCH_CHECK(
+      qa.scalar_type() == qb.scalar_type(),
+      "Add operands should have same data type.");
+}
+
+// currently we only support int8 symmetric (zero_point = 0 for inputs and output) quantized add
+template <bool kReluFused = false>
+Tensor add(Tensor qa, Tensor qb, double output_scale, int64_t output_zero_point) {
+  std::cout << "cuda add" << std::endl;
+  if (qa.numel() == 0) {
+    return Tensor{};
+  }
+  // TODO: add shape checking? I think this is contingent on whether we support broadcasting in qadd, so maybe leave this out for now?
+  check_inputs(qa, qb);
+
+  // cudnn expects tensors to be at least 3D. So we will append dummy dimensions if the input tensors are not at least 3D
+  if (qa.dim() < 3) {
+    std::vector<int64_t> new_sizes{qa.sizes().vec()};
+    while (new_sizes.size() < 3) {
+      new_sizes.emplace_back(1);
+    }
+    qa = at::native::view(qa, new_sizes);
```

Comment:
nit: I think we should be using at::view, also we should be able to do `qa.view(....)` as well I think